<launch>
  <arg name="camera_exposure_time" default="1.0"/>
  <arg name="cam_topic" default="cam_topic"/>
  <arg name="cam_trigger_topic" default="cam_trigger_topic"/>

  <arg name="camera_fps" default="30"/>

  <node name="mvsua_ros_driver" pkg="mvsua_ros_driver" type="mvsua_ros_driver" output="screen">

    <param name="camera_num" value="1"/>
    <param name="pixel_bit" type="int" value="255" />
    <param name="format" type="string" value="mono8" />
    <param name="format_pub" type="string" value="mono8" />
    <param name="format_saveshow" type="string" value="mono8" />
    <param name="show_flag" type="bool" value="false" />

    <param name="exposure_time" type="double" value="$(arg camera_exposure_time)" />

    <rosparam command="load" file="$(find mvsua_ros_driver)/config/camra_param.yaml"/>

    <remap from="~mvsua_cam/image_raw1" to="$(arg cam_topic)/image_raw" />
    <remap from="~time_synchronization_camera_msgs" to="$(arg cam_trigger_topic)" />

  </node>

  <node name="camera_trigger" pkg="mvsua_ros_driver" type="camera_trigger" output="screen">
    <param name="fps" type="int" value="$(arg camera_fps)" />
    <remap from="/time_synchronization_camera_msgs" to="$(arg cam_trigger_topic)" />
  </node>

</launch>
